src/view/components/microbit/MicrobitImage.tsx

```diff
@@ -5,6 +5,81 @@ import * as React from "react";
 import "../../styles/Microbit.css";
 import { MICROBIT_SVG } from "./Microbit_svg";
 
-export const MicrobitImage: React.FC = () => {
-    return MICROBIT_SVG;
+interface EventTriggers {
+    onMouseUp: (button: HTMLElement, event: Event, buttonKey: string) => void;
+    onMouseDown: (button: HTMLElement, event: Event, buttonKey: string) => void;
+    onMouseLeave: (
+        button: HTMLElement,
+        event: Event,
+        buttonKey: string
+    ) => void;
+}
+interface IProps {
+    eventTriggers: EventTriggers;
+    leds: number[][];
+}
+interface IState {

```

Comment:
Refs should not be a part of the state, I'm assuming the ref would stay constant throughout the lifecycle of the component